ios/browser/api/favicon/favicon_loader.h

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_FAVICON_FAVICON_LOADER_H_
+#define BRAVE_IOS_BROWSER_API_FAVICON_FAVICON_LOADER_H_
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+typedef NSInteger BraveFavIconLoaderSize NS_TYPED_ENUM
+    NS_SWIFT_NAME(FavIconLoader.Sizes);
+/// Minimal acceptable favicon size, in points.
+OBJC_EXPORT BraveFavIconLoaderSize const BraveFavIconLoaderSizeMin;
+/// Desired small favicon size, in points.
+OBJC_EXPORT BraveFavIconLoaderSize const BraveFavIconLoaderSizeDesiredSmall;
+/// Desired medium favicon size, in points.
+OBJC_EXPORT BraveFavIconLoaderSize const BraveFavIconLoaderSizeDesiredMedium;
+
+@class BraveFavIconAttributes;
+
+NS_ASSUME_NONNULL_BEGIN
+
+OBJC_EXPORT
+NS_SWIFT_NAME(FavIconLoader)
+@interface BraveFavIconLoader : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/// |sizeInPoints| - the desired size of the favIcon
+/// |minSizeInPoints| - the minimum acceptable favIcon size
+/// |completion| may be called more than once (once with a default image, and
+/// one with the actual fav-icon if found).
+- (void)favIconForPageURL:(nonnull NSURL*)url
+              sizeInPoints:(BraveFavIconLoaderSize)sizeInPoints
+           minSizeInPoints:(BraveFavIconLoaderSize)minSizeInPoints
+    fallbackToGoogleServer:(bool)fallbackToGoogleServer
+                completion:
+                    (void (^)(BraveFavIconAttributes* _Nonnull attributes))
```

Comment:
`_Nonnull` is redundant here and below because of the `NS_ASSUME_NONNULL_BEGIN`/`END`